### ISSUE TITLE
Add algebraic operations between dates

### ISSUE THREAD

**Issue by ecyshor** [crobox/clickhouse-scala-client#72]
We have a number of operations which can be executed between dates and numbers, but no support to subtract two dates. This should produce a column of int as result.

**Comment by rbrondgeest** [comment]
I do not think this is true, there are magnets for DateTime to AddSubtractable.

select(const(new DateTime()) - 1)

compiles and produces a valid qry

**Comment by ecyshor** [comment]
That is true, but there is no support to subtract two columns which represent a date (this would return the number of days between the dates). Should be pretty straightforwards to implement the magnet for localdate, localdate, int

**Comment by sjoerdmulder** [comment]
Still valid, confirmed against current main. Every `AritRetType` binding involving a date has a *numeric* right operand:

```
AritRetType[LocalDate, Int, LocalDate]   AritRetType[DateTime, Int, DateTime]
AritRetType[LocalDate, Long, LocalDate]  AritRetType[DateTime, Long, DateTime]  ... etc
```

There is no binding with a date on **both** sides, so `dateA - dateB` does not compile — matching the 2018 comment thread, which correctly distinguished date-minus-number (works) from date-minus-date (does not).

Note the related limitation found while fixing the widening table in #321: `AritRetType` is keyed only on operand types, so it cannot express that `/` always yields `Float64` while `intDiv` yields `Int64`. Adding date-minus-date returning a day count runs into the same shape — the result type depends on the operator, not just the operands — so both want addressing together.
